src/StringTools.cpp

```diff
@@ -0,0 +1,617 @@
+        StringView::const_iterator PatIt = Pattern.begin();
+        const StringView::const_iterator SrcEnd = Source.end();
+        const StringView::const_iterator PatEnd = Pattern.end();
+
+        Boole EscapedChar = false;
+        while( PatIt != PatEnd && SrcIt != SrcEnd )
+        {
+            Boole MatchResult = false;
+            if( EscapedChar ) {
+                EscapedChar = false;
+                // If we're escaped, doesn't matter what it is.  Process as normal char.
+                MatchResult = ( (*PatIt) == (*SrcIt) );
+                ++PatIt;
+                ++SrcIt;
+            }else if( (*PatIt) == '*' ) {
+                EscapedChar = false;
+                MatchResult = MatchAsterisk(SrcIt,SrcEnd,PatIt,PatEnd,CaseSensitive);
+            }else if( (*PatIt) == '[' ) {
+                EscapedChar = false;
+                MatchResult = MatchRangeCharacter(SrcIt,SrcEnd,PatIt,PatEnd);
+            }else if( (*PatIt) == '?' ) {
+                EscapedChar = false;
+                MatchResult = MatchQuestionMark(SrcIt,SrcEnd,PatIt,PatEnd);
+            }else if( (*PatIt) == '\\' ) {
+                EscapedChar = true;
+                ++PatIt;
+                continue;
+            }else{
+                EscapedChar = false;
+                if( CaseSensitive ) {
+                    MatchResult = ( (*PatIt) == (*SrcIt) );
+                }else{
+                    MatchResult = CaseInsensitiveCompare(PatIt,SrcIt);
+                }
+                ++PatIt;
+                ++SrcIt;
+            }
+
+            if( !MatchResult ) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Convert-To-Data functions
+
+    Boole ConvertToBool(const StringView ToConvert, const Boole Default)
+    {
+        auto Checker = [ToConvert](const StringView Str) {
+            if( Str.size() == ToConvert.size() ) {
+                for( size_t Idx = 0 ; Idx < Str.size() ; ++Idx )
+                {
+                    if( ::tolower( Str[Idx] ) != ::tolower( ToConvert[Idx] ) ) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            return false;
+        };
+
+        if( Checker("true") ) return true;
+        else if( Checker("yes") ) return true;
+        else if( Checker("1") ) return true;
+        else if( Checker("false") ) return false;
+        else if( Checker("no") ) return false;
+        else if( Checker("0") ) return false;
+        else return Default;
+    }
+
+    template<>
+    Int8 ConvertFromString<Int8>(const StringView ToConvert)
+    {
+        StringStream Converter;
+        int Result;
+        Converter << ToConvert;
+        Converter >> Result;
+        return static_cast<Int8>(Result);
+    }
+
+    template<>
+    UInt8 ConvertFromString<UInt8>(const StringView ToConvert)
```

Comment:
Why is this specialized?

Reply:
Implicit conversions turn it into an int, which produces unexpected results when interacting with streams.  The specializations work around that.